From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Patrick McCarty <patrick.mccarty@intel.com>
Date: Sat, 19 Jun 2021 10:56:24 -0700
Subject: [PATCH] Add makefile for easier packaging

This makefile is derived from Auke's original makefile added in
`0001-Convert-go.sh-script-into-a-classic-Makefile-with-DE.patch`, but adapted
for upcoming releases that have revised, and much simplified, the installation
procedure.

Signed-off-by: Patrick McCarty <patrick.mccarty@intel.com>
---
 Makefile | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)
 create mode 100644 Makefile

diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..ec95a1a
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,27 @@
+#
+# This makefile installs the firmware files to the place where the linux
+# firmware loading system expects them, `/usr/lib/firmware`. It supports
+# DESTDIR installations for package manager software. You should just have to
+# `sudo make install` to put the files in the correct location.
+#
+
+ifndef SOF_VERSION
+$(error SOF_VERSION must be defined in the environment or on the command line)
+endif
+
+FIRMWAREDIR ?= /usr/lib/firmware
+
+all:
+
+check:
+
+# v1.8 changes the tree structure, so support versions < v1.8 as well
+install:
+	TOPDIR=""; \
+	if [ -d v$(SOF_VERSION).x ]; then \
+		TOPDIR="v$(SOF_VERSION).x/"; \
+	fi; \
+	for DIR in sof sof-tplg; do \
+		mkdir -pv $(DESTDIR)/$(FIRMWAREDIR)/intel; \
+		cp -av $${TOPDIR}$$DIR-v$(SOF_VERSION) $(DESTDIR)/$(FIRMWAREDIR)/intel/$$DIR; \
+	done
